### DISCUSSION TITLE
Develop Website for Hasty-server

### DISCUSSION THREAD

**Discussion by IntegerAlex** [IntegerAlex/hasty-server#6] [Q&A]
Create a website/landing page for details about Framework.

**Comment by prince-63** [comment]
Please assign me this issue I love to work on it.

**Comment by IntegerAlex** [comment]
![Screenshot 2024-09-29 215621](https://github.com/user-attachments/assets/3f15530b-1172-4464-a3ea-9cb13e2ff5e3)
Refference

**Reply by IntegerAlex** [reply]
Doc reference is below

**Comment by IntegerAlex** [comment]
# Hasty Server

Hasty Server is a lightweight web framework for Node.js inspired by Express.js. The key difference is that instead of using the `http` module from Node.js, Hasty Server implements the HTTP protocol using raw TCP sockets through the `net` library. This framework is designed for developers who want to dive deeper into how HTTP works under the hood and learn how to build their own web servers.

## Features

- **Raw TCP Socket Handling**: Implements the HTTP protocol using the `net` library instead of Node.js's `http` module.
- **Custom Request Parser**: Parses HTTP requests, including headers and bodies, allowing developers to see how HTTP is structured.
- **Flexible Routing**: Simple routing system to handle different HTTP methods like `GET` and `POST`.
- **JSON Body Handling**: Provides basic JSON body parsing to work with API requests.
- **Easy to Extend**: Allows developers to add more features and customize how HTTP requests and responses are handled.
- **Lightweight**: Minimal dependencies, focusing on simplicity and clarity.

## Installation

```bash
npm install hasty-server
```
## Quick Start 

```javascript
const Hasty = require('hasty-server');
const server = new Hasty();

server.get('/', (req, res) => {
    res.send('Hello World');
});

server.listen(8080, () => {
    console.log('Server is running on port 8080');
});
```

## Key Concepts
Routing: Hasty Server provides basic routing functionality. Use server.get() or server.post() to handle specific routes.
```javascript
server.get('/home', (req, res) => {
    res.send('Welcome to the Home Page');
});
```
Request Parsing: Requests are parsed manually, allowing developers to learn the structure of HTTP requests including methods, paths, and bodies.

Response Handling: Send responses using the res.send() method or return a status code with res.sendStatus().

```javascript
server.post('/data', (req, res) => {
    res.send(JSON.stringify({ message: 'Data received!' }));
});
```

**Comment by IntegerAlex** [comment]
![HASTY](https://github.com/user-attachments/assets/1b071005-8155-4434-bf91-3492f9c6a183)

logo

**Comment by prince-63** [comment]
Ok I will add this also in website.
